addons/gamification/models/gamification_karma_tracking.py

```diff
@@ -5,65 +5,90 @@
 
 from dateutil.relativedelta import relativedelta
 
-from odoo import api, fields, models
+from odoo.tools import date_utils
+from odoo import api, fields, models, _
 
 
 class KarmaTracking(models.Model):
     _name = 'gamification.karma.tracking'
     _description = 'Track Karma Changes'
     _rec_name = 'user_id'
-    _order = 'tracking_date DESC'
+    _order = 'tracking_date desc, id desc'
 
-    user_id = fields.Many2one('res.users', 'User', index=True, readonly=True, required=True, ondelete='cascade')
-    old_value = fields.Integer('Old Karma Value', required=True, readonly=True)
-    new_value = fields.Integer('New Karma Value', required=True, readonly=True)
+    user_id = fields.Many2one('res.users', 'User', index=True, required=True, ondelete='cascade')
+    old_value = fields.Integer('Old Karma Value', compute='_compute_old_value', store=True, group_operator="min")
+    new_value = fields.Integer('New Karma Value', required=True, group_operator="max")
     consolidated = fields.Boolean('Consolidated')
-    tracking_date = fields.Date(default=fields.Date.context_today)
+    # tracking_date is used as create_date would not make the job (take another date as the current date and time)
+    # when creating the monthly consolidated data. id of the record will be jointly used as milliseconds are not stored
+    # in Datetime fields
+    tracking_date = fields.Datetime(default=fields.Datetime.now, readonly=True, index=True)
+    reason = fields.Text(default='')
+    originated_by = fields.Text(compute='_compute_originated_by', store=True)
+    origin_type = fields.Text(compute='_compute_origin_type', default="Manual Entry", store=True)
```

Comment:
Because TDE was not for it ;o)
He preferred this implementation

Reply:
well, if the number of possible values are limited, better go for a selection field.
And if someone would need to change manually the karma of a user, we could use a type 'manual' (if origin_id is not set for example) -> In this case, reason could stay a text field.
To check

Reply:
@tde-banana-odoo can you advise?

Reply:
My appologies... I thought you were talking about something else... a reference field for the link with the website_forum.post or website_slides.slides...

So you are totally right :) I'll check that up!

Reply:
@tde-banana-odoo : regarding ref field instead of 3 fields that represent the same thing (origin_id).
I personally think that would really lighthen the code.
@IT-Ideas tells me that you already discussed about that and you choose to go for multiple fields.
Can you reconsider ? :)
Muchas Bisoutas :kissing_cat:

Reply:
I don't remember all details to be honest ... anyway now that the feature is completely implemented if it is easier with a reference field you can go for it :) . That's the purpose of review, simplifying code !

What we need: be able to jump directly on source record (m2o link), maybe also group_by (does this work with a reference field ?). Like if I want to know who gained karma from my "HyperQuizz" slide, because I have doubts about it for example.